exercism/elixir/twelve-days/twelve_days.exs

```diff
@@ -45,9 +45,8 @@ defmodule TwelveDays do
     verses(1, 12)
   end
 
-  defp gifts(0), do: []
-  defp gifts(number) do
-    gift = @translation[number] |> elem(1)
-    [gift | gifts(number - 1)]
+  def gifts(number) do
+    1..number
+    |> Enum.reduce([], fn n, acc -> [@translation[n] |> elem(1) | acc] end)
```

Comment:
Why not just use map??
`number..1 |> Enum.map(fn n -> @translation[n] |> elem(1) end)`
or
`number..1
|> Stream.map(fn x -> @translation[x] end)
|> Enum.map(fn {number, gift} -> gift end)
`
Your pattern matching solution is also not tail recursive, you could refactor it to make it tail recursive, but I believe the readability would go down.
`  defp gifts(0, acc), do: Enum.reverse(acc)`
  `defp gifts(n, acc) do
    gifts(n-1, [@translation[n] |> elem(1) | acc])
  end
`